Fix sorting of dictionaries for asg tags

KaloBeshWork opened the pull request:
### What does this PR do?
In Python 3 dictionaries do not have ordering defined, which may result in an exception when sorting ASG tags. This fix aims to resolve that issue.
File that was edited: salt/utils/boto3mod.py

### What issues does this PR fix or reference?
Fixes: #52404
https://github.com/saltstack/salt/issues/52404

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltstack.com/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

garethgreenaway committed:
Adding the ability to override the return code based on either stdout or stderr output.

garethgreenaway committed:
Fixing lint

garethgreenaway committed:
Fixing test_retcode_with_success_retcodes to be more Windows friendly.

cbosdo committed:
Don't call zypper with more than one --no-refresh

Now zypper started being picky and errors out when --no-refresh is
passed twice. Make sure we won't hit this.

garethgreenaway committed:
Porting PR #49133 to 2019.2.1

garethgreenaway committed:
Adding test for returner function in postgres_local_cache returner.

garethgreenaway committed:
Fixing broken tests.

garethgreenaway committed:
Removing version restraint on installing the mock package.  Fixing postgres_local_cace returner tests.  Updating the test_returner_unicode_exception test to ensure we are triggering the UnicodeDecodeError exception. Also ensuring that we are using the sme types when looking for the return in the dictionary string.

garethgreenaway committed:
Only encode if we are running on Python2.

Commit:
add tiamat run.py script

Commit:
Added changelog entry

major0 committed:
feat: added `allow_one_of()` and `require_one_of()` decorators

- `allow_one_of()` raises an error if more than one of the allowed
  parameters is supplied.
- `require_one_of()` operates the same as `allow_one_of()` but also
  raises an error if none of the allowed parameters are supplied.

jfindlay committed:
salt.returners demote log statements to debug

These logs tend to produce a great amount of low value log records which
would not be useful unless you were debugging returner code.  For
example:

```
2020-10-12 12:37:04,228 [salt.returners   :166 ][INFO    ] Using default for db_returner db_path
2020-10-12 12:37:04,232 [salt.returners   :166 ][INFO    ] Using default for db_returner ignored_retcodes
2020-10-12 12:37:04,235 [salt.returners   :166 ][INFO    ] Using default for db_returner db_path
2020-10-12 12:37:04,246 [salt.returners   :166 ][INFO    ] Using default for db_returner ignored_retcodes
2020-10-12 12:37:04,248 [salt.returners   :166 ][INFO    ] Using default for db_returner db_path
2020-10-12 12:37:04,252 [salt.returners   :166 ][INFO    ] Using default for db_returner ignored_retcodes
2020-10-12 12:37:04,255 [salt.returners   :166 ][INFO    ] Using default for db_returner db_path
2020-10-12 12:37:04,265 [salt.returners   :166 ][INFO    ] Using default for db_returner ignored_retcodes
2020-10-12 12:37:04,267 [salt.returners   :166 ][INFO    ] Using default for db_returner db_path
2020-10-12 12:37:04,270 [salt.returners   :166 ][INFO    ] Using default for db_returner ignored_retcodes
2020-10-12 12:37:04,272 [salt.returners   :166 ][INFO    ] Using default for db_returner db_path
```

acgustafson committed:
Change wget download name to match tar command below it

jfindlay committed:
salt.returners Drop six usage and Py2 support

kgbsd committed:
runas user for FreeBSD should be 'postgres' not pgsql

ioncebotari committed:
Added support for openeuler and issabel PBX

ioncebotari committed:
Fix for ci-precommit check

ioncebotari committed:
Second fix for ci-precommit check

ioncebotari committed:
Added changelog for the fix

dehnert committed:
Note security concerns with grains for targeting

Given that Vault is used for managing secrets, it seems useful to remind
people that grains are generally minion-controlled when talking about
using them to assign policies (and consequently give access to secrets).
This is related to #43287, though only warns people of the issue, rather
than resolving it by adding (eg) Pillar-based targeting.

This change also cleans up some nearby style and formatting issues with
the docs.

meaksh committed:
Fix aptpkg.normalize_name when package arch is 'all'

meaksh committed:
Add test case of DEB package where arch is 'all'

meaksh committed:
Do not crash when unexpected cmd output at listing patches

meaksh committed:
Add unit tests to cover unexpected output when listing patches

meaksh committed:
Add changelog

meaksh committed:
Add changelog

cbosdo committed:
virt: convert spice generation tests to pytests

cbosdo committed:
virt: don't add spicevmc channel to Xen VMs

Xen fully virtualized VMs with spicevmc channel fail to start, so better
not write it out in such cases.

cbosdo committed:
virt: inverse the remaining asserts in pytests

Commit:
Fixed zabbix mod get connection data from pillar

myii committed:
test(test_avahi_announce): convert to pytest

myii committed:
test(test_bonjour_announce): convert to pytest

myii committed:
test(test_haproxy): convert to pytest

krionbsd committed:
Move cmd state test to pytest

krionbsd committed:
Fix typo and remove superfluous comment

krionbsd committed:
Fix typo

myii committed:
test(test_adb): convert to pytest

myii committed:
test(test_glxinfo): convert to pytest

myii committed:
test(test_log_beacon): convert to pytest

krionbsd committed:
s/command/comment

ggiesen committed:
Fix link to Doug Hellmann's blog entry on dictionary creation

krionbsd committed:
Move NX-OS state tests to pytest

xuchunmei000 committed:
add aliyun linux support and set alinux as redhat family

Signed-off-by: Chunmei Xu <xuchunmei@linux.alibaba.com>

krionbsd committed:
Reformat docstrings

ggiesen committed:
Update states.firewalld docs with rich rules example. Resolves #58360

sagetherage committed:
Update issue templates

Adding `needs-traige` label to Bug, Docs, Feature and adding Test Failure template

krionbsd committed:
Move xmpp state tests to pytest

krionbsd committed:
Move xml related state tests to pytest

krionbsd committed:
Move htpasswd file related state tests to pytest

krionbsd committed:
Move vbox guest related state tests to pytest

krionbsd committed:
Remove blank lines

krionbsd committed:
Move user state tests to pytest

krionbsd committed:
Move tomcat state tests to pytest

krionbsd committed:
Remove superfluous quotes

ggiesen committed:
pre-commit changes for firewalld

krionbsd committed:
Move multiple tests to pytest

krionbsd committed:
Move multiple tests to pytest

KaloBeshWork committed:
In Python 3 dictionaries do not have ordering defined, which may result in an exception when sorting ASG tags. This fix aims to fix that.

File that was edited: salt/utils/boto3mod.py
Fixes: #52404

https://github.com/saltstack/salt/issues/52404

krionbsd commented:
@KaloBeshWork thanks, could you please fix lint error, we also require a text case for this change in `tests/unit/utils/test_botomod.py`

KaloBeshWork committed:
In Python 3 dictionaries do not have ordering defined, which may result in an exception when sorting ASG tags. This fix aims to fix that.

File that was edited: salt/utils/boto3mod.py
Fixes: #52404

https://github.com/saltstack/salt/issues/52404

KaloBeshWork committed:
Wrote test

KaloBeshWork committed:
Fixed lint error

KaloBeshWork committed:
Ran Pre-Commit

KaloBeshWork committed:
Fixes

krionbsd committed:
Move remaining PostgreSQL state tests to pytest.
Get rid of duplicate tests in PostgreSQL extensions and schemas.

krionbsd committed:
Move MySQL state tests to pytest

krionbsd committed:
Remove redundant brackets

krionbsd committed:
Fix docstrings formatting

krionbsd committed:
Fix typo in docstrings

Ch3LL committed:
Remove glance state module in favor of glance_image

ggiesen committed:
Correct link to napalm installation. Resolves #56501

ggiesen committed:
Commit with pre-commit enabled to catch any formatting changes.

ggiesen committed:
Correct link to napalm installation. Resolves #56512

ggiesen committed:
Correct link to napalm installation. Resolves #56511

ggiesen committed:
Commit with pre-commit enabled to catch any formatting changes.

twangboy committed:
Move to pytest, fix typo

eliasp committed:
modules/match: add missing "minion_id" in Pillar example

The documented Pillar example for `match.filter_by` lacks the `minion_id` parameter. Without it, the assignment won't work as expected.
- fix documentation
- add tests:
  - to prove the misbehavior of the documented example
  - to prove the proper behaviour when supplying `minion_id`
  - to ensure some misbehaviour observed with compound matchers doesn't occur

krionbsd committed:
Add missing word in syndic docs

garethgreenaway committed:
bump deprecation warning to Silicon.

ScriptAutomate committed:
Update Sphinx dep tree

ScriptAutomate committed:
Fix raw code block errors from pygments

ScriptAutomate committed:
Updating default fallback for latest_release

s0undt3ch committed:
Use the checkout major version as the latest release

krionbsd committed:
Revert changes in salt.7

krionbsd committed:
Pack MySQL related tests into own directory.
Use pytest.importorskip

KaloBeshWork committed:
Fixes

KaloBeshWork committed:
Fixes

Ch3LL committed:
update wording in changelog

krionbsd committed:
Rename tests

garethgreenaway committed:
Updating warnutil version to Phosphorous.

krionbsd committed:
Add "poudriere -i -j name" option to list jail information for
poudriere.
Move test to pytest.

krionbsd committed:
Add changelog

twangboy committed:
Handle gpt.ini with unix line endings

ScriptAutomate committed:
Add back version var used by Sphinx

ScriptAutomate committed:
Remove conditional or from latest release var

ScriptAutomate committed:
Update docs deps to include linux deps

ScriptAutomate committed:
Update docs to use pkg linux reqs

ScriptAutomate committed:
Regen/reset doc dep tree

ScriptAutomate committed:
Update nox to install only docs.txt

twangboy committed:
Add tests, ensure windows-style line endings

ScriptAutomate committed:
Adding linux.txt pkg req files for triggers

ScriptAutomate committed:
linux.in instead of common.in for docs deps

ScriptAutomate committed:
Ensure base upgrades in docs venv before run

KaloBeshWork committed:
Fixes

Commit:
Two initial pytests for Solaris 11 IPS packaging

lorengordon committed:
Lints backport with new rules

lorengordon committed:
Replaces Neon references with Silicon

lorengordon committed:
Adds changelog entry for success_stdout and success_stderr arguments

garethgreenaway committed:
Update salt/modules/keystone.py

Co-authored-by: Megan Wilhite <megan.wilhite@gmail.com>

Commit:
Further IPS tests for Python3 Solaris 11.4

Commit:
Additional tests for IPS

krionbsd committed:
Move apache, zabbix, postgresql and rabbitmq state tests into own
directories.

krionbsd committed:
Add more pkgng tests.

Commit:
Fix private network conversion

Commit:
Add changelog

Commit:
Lint

Commit:
Lint

keslerm committed:
Adding mendel linux

Commit:
add distro_arch

Commit:
Altered imprt pytest as per reviewer's comments

dwoz committed:
Salt api honors --log-file correctly

Make sure we verify the proper path when the --log-file cli options is
passed via the commandline. Fixes #59880

Commit:
update setup.py

Ch3LL committed:
Parameterize test_pillar_match_filter_by_minion_id test

dwoz committed:
Merge 3003 changes forward to the master branch (#59879)

* Merge 3002.6 bugfix changes (#59822)

* Pass `CI_RUN` as an environment variable to the test run.

This allows us to know if we're running the test suite under a CI
environment or not and adapt/adjust if needed

* Migrate `unit.setup` to PyTest

* Backport ae36b15 just for test_install.py

* Only skip tests on CI runs

* Always store git sha in _version.py during installation

* Fix PEP440 compliance.

The wheel metadata version 1.2 states that the package version MUST be
PEP440 compliant.

This means that instead of `3002.2-511-g033c53eccb`, the salt version
string should look like `3002.2+511.g033c53eccb`, a post release of
`3002.2` ahead by 511 commits with the git sha `033c53eccb`

* Fix and migrate `tests/unit/test_version.py` to PyTest

* Skip test if `easy_install` is not available

* We also need to be PEP440 compliant when there's no git history

* Allow extra_filerefs as sanitized kwargs for SSH client

* Fix regression on cmd.run when passing tuples as cmd

Co-authored-by: Alexander Graul <agraul@suse.com>

* Add unit tests to ensure cmd.run accepts tuples

* Add unit test to check for extra_filerefs on SSH opts

* Add changelog file

* Fix comment for test case

* Fix unit test to avoid failing on Windows

* Skip failing test on windows

* Fix test to work on Windows

* Add all ssh kwargs to sanitize_kwargs method

* Run pre-commit

* Fix pylint

* Fix cmdmod loglevel and module_names tests

* Fix pre-commit

* Skip ssh tests if binary does not exist

* Use setup_loader for cmdmod test

* Prevent argument injection in restartcheck

* Add changelog for restartcheck fix

* docs_3002.6

* Add back tests removed in merge

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>
Co-authored-by: Megan Wilhite <megan.wilhite@gmail.com>
Co-authored-by: Bryce Larson <brycel@vmware.com>
Co-authored-by: Pablo Suárez Hernández <psuarezhernandez@suse.com>
Co-authored-by: Alexander Graul <agraul@suse.com>
Co-authored-by: Frode Gundersen <fgundersen@saltstack.com>

* Remove glance state module in favor of glance_image

* update wording in changelog

* bump deprecation warning to Silicon.

* Updating warnutil version to Phosphorous.

* Update salt/modules/keystone.py

Co-authored-by: Megan Wilhite <megan.wilhite@gmail.com>

* Check $HOMEBREW_PREFIX when linking against libcrypto

When loading `libcrypto`, Salt checks for a Homebrew installation of `openssl`
at Homebrew's default prefix of `/usr/local`. However, on Apple Silicon Macs,
Homebrew's default installation prefix is `/opt/homebrew`. On all platforms,
the prefix is configurable.  If Salt doesn't find one of those `libcrypto`s,
it will fall back on the un-versioned `/usr/lib/libcrypto.dylib`, which will
cause the following crash:

    Application Specific Information:
    /usr/lib/libcrypto.dylib
    abort() called
    Invalid dylib load. Clients should not load the unversioned libcrypto dylib as it does not have a stable ABI.

This commit checks $HOMEBREW_PREFIX instead of hard-coding `/usr/local`.

* Add test case

* Add changelog for 59808

* Add changelog entry

* Make _find_libcrypto fail on Big Sur if it can't find a library

Right now, if `_find_libcrypto` can't find any externally-managed versions of
libcrypto, it will fall back on the pre-Catalina un-versioned system libcrypto.
This does not exist on Big Sur and it would be better to raise an exception
here rather than crashing later when trying to open it.

* Update _find_libcrypto tests

This commit simplifies the unit tests for _find_libcrypto by mocking out the
host's filesystem and testing the common libcrypto installations (brew, ports,
etc.) on Big Sur. It simplifies the tests for falling back on system versions
of libcrypto on previous versions of macOS.

* Fix description of test_find_libcrypto_with_system_before_catalina

* Patch sys.platform for test_rsax931 tests

* modules/match: add missing "minion_id" in Pillar example

The documented Pillar example for `match.filter_by` lacks the `minion_id` parameter. Without it, the assignment won't work as expected.
- fix documentation
- add tests:
  - to prove the misbehavior of the documented example
  - to prove the proper behaviour when supplying `minion_id`
  - to ensure some misbehaviour observed with compound matchers doesn't occur

* Fix for issue #59773

- When instantiating the loader grab values of grains and pillars if
  they are NamedLoaderContext instances.
- The loader uses a copy of opts.
- Impliment deepcopy on NamedLoaderContext instances.

* Add changelog for #59773

* _get_initial_pillar function returns pillar

* Fix linter issues

* Clean up test

* Bump deprecation release for neutron

* Uncomment Sulfur release name

* Removing the _ext_nodes deprecation warning and alias.

* Adding changelog.

* Renaming changelog file.

* Update 59804.removed

* Initial pass at fips_mode config option

* Fix pre-commit

* Fix tests and add changelog

* update docs 3003

* update docs 3003 - newline

* Fix warts in changelog

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>
Co-authored-by: Megan Wilhite <megan.wilhite@gmail.com>
Co-authored-by: Bryce Larson <brycel@vmware.com>
Co-authored-by: Pablo Suárez Hernández <psuarezhernandez@suse.com>
Co-authored-by: Alexander Graul <agraul@suse.com>
Co-authored-by: Frode Gundersen <fgundersen@saltstack.com>
Co-authored-by: Gareth J. Greenaway <gareth@saltstack.com>
Co-authored-by: Gareth J. Greenaway <gareth@wiked.org>
Co-authored-by: Hoa-Long Tam <hoalong@apple.com>
Co-authored-by: krionbsd <krion@freebsd.org>
Co-authored-by: Elias Probst <e.probst@ssc-services.de>
Co-authored-by: Frode Gundersen <frogunder@gmail.com>

Commit:
add ubuntu-2004-amd64 m2crypto pycryptodome and tcp tests

Commit:
update run.py to latest from https://gitlab.com/saltstack/open/salt-pkg/-/raw/master/pkgs/redhat/sources/run.py

Commit:
Accept zabbix.xyz or zabbix:xyz as connection data

twangboy committed:
Remove logfile check

krionbsd committed:
Add AArch64 support

max-arnold committed:
Prune obsolete exception handler in dictupdate.update

krionbsd committed:
Add platform.machine() to get arch properly

krionbsd committed:
Add changelog entry

krionbsd committed:
Add changelog

twangboy committed:
Try monkeypatch

twangboy committed:
Fix logging

twangboy committed:
Remove unused import

Commit:
Added grains.uuid on Windows

twangboy committed:
Revert log factory after test run

Commit:
Added changelog entry

krionbsd committed:
Skip tests on FreeBSD because of timeouts issue

krionbsd committed:
Add comma

twangboy committed:
Just check for bytes

Ch3LL committed:
Add changelog for 58915

krionbsd committed:
Changelog rewording

Ch3LL committed:
Add changelog for 46942

Ch3LL committed:
run pre-commit

Commit:
add run.py to the sdist tarball

vin01 committed:
Always honor state_type when compiling require_in

garethgreenaway committed:
Update tests/unit/returners/test_postgres_local_cache.py

Committing requested changes.

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>

garethgreenaway committed:
Update tests/unit/returners/test_postgres_local_cache.py

Committing requested changes.

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>

garethgreenaway committed:
Update test_postgres_local_cache.py

garethgreenaway committed:
Update test_postgres_local_cache.py

Putting this back in.

lorengordon committed:
Updates docstring and versionadded per requested change in review

Commit:
init changes from dmitry PR

xeacott committed:
pre-commit

xeacott committed:
add changelog file

ggiesen committed:
Fix typo in Network Automation docs, fixes #59948

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

xeacott committed:
Fix 20789 - Window's user creation and password policy requirements (#59563)

* surface reason why user could not be created

* init commit before tests

* add changelog file

* bruh

* fix docstring

* better handle on logic

* convert first test to pytest

* pre-commit

* add runs_on to test

* fix lint

* more fixes

* fix nix tests

* fix windows tests

* add extra check to not break other tests

* pre commit

* remove extra check

Co-authored-by: Joe Eacott <jeacott@vmware.com>
Co-authored-by: Sage the Rage <36676171+sagetherage@users.noreply.github.com>

lorengordon committed:
Simplifies changelog entry

lorengordon committed:
Simplifies assignment of success_stdout and success_stderr

lorengordon committed:
Supports matching partial strings against stderr and stdout

Ch3LL committed:
Update __utils__ calls to import utils in azure

ggiesen committed:
Rework documentation for python3 version of jupyter console (which has been split out from IPython). Fixes #59741

ggiesen committed:
Remove errant '?'

ggiesen committed:
Change debian python3-jupyter-console to jupyter-console to get the front-end script (which will also install python3-jupyter-console lib)

vin01 committed:
Simplify state_type matching

bergentroll committed:
Add grains fix for Astra Linux (#59353)

Ch3LL committed:
Add changelog for 59744

Ch3LL committed:
Fix azure unit tests and move to pytest

frogunder committed:
fix 59946

ggiesen committed:
Fix openconfig links

ggiesen committed:
Revert "Change debian python3-jupyter-console to jupyter-console to get the front-end script (which will also install python3-jupyter-console lib)"

This reverts commit 4d6b15a70ec16bed8e2d08ddbe32f2e750186155.

ggiesen committed:
Revert "Remove errant '?'"

This reverts commit 546a8592e0c052e7aa5dff844099caffeb73f42b.

ggiesen committed:
Revert "Rework documentation for python3 version of jupyter console (which has been split out from IPython). Fixes #59741"

This reverts commit 69764e3f788564dac66845b39f0936e2ca7aa617.

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

Commit:
Change naming according to hetzner doc

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

Commit:
Fix getting the right attribuyte vom hetzner private net

Commit:
Test public_ip and private_net convert to dict functions

KaloBeshWork committed:
Fixes

KaloBeshWork committed:
Fixes

KaloBeshWork committed:
Merge remote-tracking branch 'origin/Fix_sorting_of_dictionaries_to_sort_ASG_tags' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

StackKorora committed:
Adding preliminary support for Rocky. (#59682)

* Adding preliminary support for Rocky.

* Adding changelog and test per MR guidence.

* Update test_core.py

Fix a clean up issue

Co-authored-by: Megan Wilhite <megan.wilhite@gmail.com>
Co-authored-by: Gareth J. Greenaway <gareth@wiked.org>

Commit:
Fix for Solaris 10 pkg.install issue 59305 and tests

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

dmurphy18 committed:
Update salt/modules/solarispkg.py

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>

Commit:
Skip '-t' type for zfs get if Solaris as not valid parameter on that platform

Commit:
Replaced versionedited with versionchanged

Commit:
changing the cloud platforms file missed in 24b69a9e389939d543b2b2b98318820da88ece66

KaloBeshWork committed:
Removed " cicd/jenkins/kitchen-macosxcatalina-py3-pytest "

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

vin01 committed:
Add changelog for #59922

Commit:
Added unit tests for ZFS on Solaris unsupported '-t' type

Commit:
Updated to allow for reviewer comments

KaloBeshWork committed:
Error in Jenkins test: pr-ubuntu-2004-amd64-py3-tcp-pytest

KaloBeshWork committed:
Error in Jenkins test: pr-ubuntu-2004-amd64-py3-tcp-pytest

KaloBeshWork committed:
Returned previous version

KaloBeshWork committed:
Returned previous version

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge remote-tracking branch 'origin/Fix_sorting_of_dictionaries_to_sort_ASG_tags' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork commented:
Hi,

There is a test that most of the time fails(**pr-macosx-catalina-x86_64-py3-pytest**), but it runs fine on my machines(video below for reference). Is there something I am missing?

Thank you. 

[](https://youtu.be/McfnU08Edtk)

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Small change to .gitignore

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Added entry to changelog

KaloBeshWork committed:
Merge remote-tracking branch 'origin/Fix_sorting_of_dictionaries_to_sort_ASG_tags' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Changed log warning when sorting

KaloBeshWork committed:
Changed log warning when sorting

KaloBeshWork committed:
Merge remote-tracking branch 'origin/Fix_sorting_of_dictionaries_to_sort_ASG_tags' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

KaloBeshWork committed:
Merge branch 'master' into Fix_sorting_of_dictionaries_to_sort_ASG_tags

dwoz commented:
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.
